common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java

```diff
@@ -0,0 +1,175 @@
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContract;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.util.StringUtil;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+/**
+ * Base activity used to launch a Install WPJ Certificate request,
+ * and get the result from the Activity using the API {@link AppCompatActivity#registerForActivityResult(ActivityResultContract, ActivityResultCallback)}
+ */
+@Accessors(prefix = "m")
+public final class InstallCertActivityLauncher extends AppCompatActivity {
+    private static final String TAG = InstallCertActivityLauncher.class.getSimpleName();
+    private static final String INSTALL_CERT_INTENT_STARTED = "broker_intent_started";
+    public static final String INSTALL_CERT_INTENT = "install_cert_intent";
+    private Intent mInstallCertificateIntent;
+    private Boolean mInstallCertificateIntentStarted = false;
+    private Boolean mInstallCertificateResultReceived = false;
+
+    @Setter(AccessLevel.PROTECTED)
+    private static IInstallCertCallback mCallback;
+    @Setter(AccessLevel.PROTECTED)
+    private static String mResultKey;
+    @Setter(AccessLevel.PROTECTED)
+    private static String mErrorKey;
```

Comment:
Should we rather provide a builder and make the class immutable?

Reply:
remove static variables.